src/api/plugins/tdbp/warren_tdbp/models.py

```diff
@@ -0,0 +1,90 @@
+"""Warren TdBP indicator models ."""
+
+from datetime import date
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel
+
+
+class Activities(Enum):
+    """Activities identifiers."""
+
+    ASSIGNMENT_SUBMITTED = "\\mod_assign\\event\\assessable_submitted"
+    ASSIGNMENT_GRADED = "\\mod_assign\\event\\submission_graded"
+    FEEDBACK = "\\mod_feedback\\event\\response_submitted"
+    FORUM_DISCUSSION_CREATED = "\\mod_forum\\event\\discussion_created"
+    FORUM_POST_CREATED = "\\mod_forum\\event\\post_created"
+    TEST = "\\mod_quiz\\event\\attempt_submitted"
+    SCORM_PACKAGE_LAUNCHED = r"\mod_scorm\event\sco_launched"
+    SCORM_PACKAGE_RAW_SUBMITTED = r"\mod_scorm\event\scoreraw_submitted"
+    SCORM_PACKAGE_STATUS_SUBMITTED = r"\mod_scorm\event\status_submitted"
+
+
+class Ressources(Enum):
+    """Ressources identifiers."""
+
+    BOOK = "\\mod_book\\event\\chapter_viewed"
+    CHAT = r"\mod_chat\event\course_module_viewed"
+    DATABASE = "\\mod_data\\event\\course_module_viewed"
+    FOLDER = "\\mod_folder\\event\\course_module_viewed"
+    FORUM = "\\mod_forum\\event\\discussion_viewed"
+    GLOSSARY = "\\mod_glossary\\event\\course_module_viewed"
+    IMS_CONTENT_PACKAGE = "\\mod_imscp\\event\\course_module_viewed"
+    EXTERNAL_TOOL = "\\mod_lti\\event\\course_module_viewed"
+    PAGE = "\\mod_page\\event\\course_module_viewed"
+    URL = "\\mod_url\\event\\course_module_viewed"
+    WIKI = "\\mod_wiki\\event\\course_module_viewed"
+
+
+class Window(BaseModel):
+    """Model for sliding window dates."""
+
+    since: date
+    until: date
+
+
+class Action(BaseModel):
+    """Model for a computed active action.
+
+    Attributes:
+        iri (str): action identifier.
+        module_type (str):
+        activation_date (str): day when the action has been activated.
+        cohort (list): IDs of student who made this action in the frame of the
+            sliding window.
+        activation_rate (float): Ratio of the student compared to the dynamic cohort
+            who made this action.
+    """
+
+    iri: str
+    name: str
+    module_type: Union[Ressources, Activities]
+    activation_date: date
+    activation_rate: float
+    activation_students: List[str]
+
+
+class SlidingWindowIndicator(BaseModel):
```

Comment:
I think the opposite: this is the sliding window result, not the indicator itself!

```suggestion
class SlidingWindow(BaseModel):
```